components/scream/src/share/tests/common_physics_functions_tests.cpp

```diff
@@ -219,6 +225,32 @@ void run(std::mt19937_64& engine)
   REQUIRE( Check::equal(PF::calculate_dz(ScalarT(p0),ScalarT(p0),one,zero),ScalarT(Rd/g)) );
   REQUIRE( Check::approx_equal(PF::calculate_dz(ScalarT(g),ScalarT(Rd),T0,zero),T0,test_tol) );
 
+  // MMR and VMR property tests:
+  //  - calculate_vmr_from_mmr(mmr=0) = 0
+  //  - calculate_vmr_from_mmr(gas_name="h2o",mmr=0.5) = 1/ep_2
+  //  - calculate_vmr_from_mmr(gas_name="dummy") should throw an error
+  //  - calculate_mmr_from_vmr(vmr=0) = 0
+  //  - calculate_mmr_from_vmr(gas_name="h2o",1.0) = ep_2/(1+ep_2)
+  //  - calculate_mmr_from_vmr(gas_name="dummy") should throw an error
+  //  - calculate_vmr_from_mmr(calculate_mmr_from_vmr(gas_name="h2o",vmr0)) = vmr
+  //  - calculate_vmr_from_mmr(calculate_mmr_from_vmr(gas_name="o2",vmr0)) = vmr, test that changing gas name changes the result.
+  //  - calculate_mmr_from_vmr(calculate_vmr_from_mmr(gas_name="h2o",mmr0)) = vmr
+  //  - calculate_mmr_from_vmr(calculate_vmr_from_mmr(gas_name="o2",mmr0)) != vmr, test that changing gas name changes the result.
+  mmr0 = pdf_mmr(engine);
+  vmr0 = pdf_mmr(engine);
+  REQUIRE( Check::equal(PF::calculate_vmr_from_mmr("h2o",zero),zero) );
+  REQUIRE( Check::approx_equal(PF::calculate_vmr_from_mmr("h2o",0.5),1.0/ep_2,test_tol) );
+  REQUIRE_THROWS( PF::calculate_vmr_from_mmr("dummy",zero) );
+  REQUIRE( Check::equal(PF::calculate_mmr_from_vmr("h2o",zero),zero) );
+  REQUIRE( Check::approx_equal(PF::calculate_mmr_from_vmr("h2o",1.0),ep_2/(1.0+ep_2),test_tol) );
+  REQUIRE_THROWS( PF::calculate_mmr_from_vmr("dummy",zero) );
+  tmp = PF::calculate_vmr_from_mmr("h2o",mmr0);
+  REQUIRE( Check::approx_equal(PF::calculate_mmr_from_vmr("h2o",tmp),mmr0,test_tol) );
+  REQUIRE( !Check::approx_equal(PF::calculate_mmr_from_vmr("o2",tmp),mmr0,test_tol) );
```

Comment:
I'm not sure that testing that mmr("o2")!=mmr("h2o") adds a lot of value. What you could do is testing that the fcn scales as expected. That is, check that compute_mmr(2X) = alpha*compute_mmr(X), for the proper alpha (and within rounding errors). Or maybe check that `vmr_from_mmr("h20",mmr)/vmr_from_mmr("o2",mmr)` is equal to the two gases molar mass ratio.